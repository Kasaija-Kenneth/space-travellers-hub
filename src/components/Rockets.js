import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rockets';
import RocketsList from './RocketItems';

const Rockets = () => {
  const dispatch = useDispatch();
  const { loading, Rockets } = useSelector((state) => state.rockets);

  useEffect(() => {
    if (Rockets.length === 0) {
      dispatch(fetchRockets());
    }
  }, [dispatch]);

  return (

    <div>
      {loading && <div>Loading ...</div>}
      {!loading && (<RocketsList />)}

    </div>
  );
};

export default Rockets;
