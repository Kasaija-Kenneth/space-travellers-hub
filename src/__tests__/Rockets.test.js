import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rockets from '../components/Rockets';

describe('snapshots testing', () => {
  test('snapshots for Rockets component', () => {
    const rendererComponents = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponents).toMatchSnapshot();
  });
});
