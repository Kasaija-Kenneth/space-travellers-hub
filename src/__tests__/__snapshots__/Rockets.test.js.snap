// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots testing snapshots for Rockets component 1`] = `
<div>
  <div />
</div>
`;
